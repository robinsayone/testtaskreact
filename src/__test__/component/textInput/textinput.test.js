import React from "react";
import reactDom from "react-dom";
import { render,screen , cleanup ,fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';
import TextInput from './../../../component/textInput/textinput';

afterEach(cleanup);

it("textinput render without crashing",()=>{
    const div = document.createElement("div");
    reactDom.render(<TextInput></TextInput>,div);
});
it("textinput renders correctly" , () =>{
    render(<TextInput label={"Email"}></TextInput>);
    expect(screen.getByLabelText(/Email/i)).toBeInTheDocument();

});

it("textinput onchange works correctly",()=>{
    const samefun = jest.fn();
    render(<TextInput label={"Email"} onChange={samefun}></TextInput>);
    const element = screen.getByTestId("textinputTextId").querySelector('input');
    fireEvent.change(element,{target:{value:'test'}});          

});

it("textinput matches snapshot",()=>{
    const tree = renderer.create(<TextInput></TextInput>).toJSON();
    expect(tree).toMatchSnapshot();
});