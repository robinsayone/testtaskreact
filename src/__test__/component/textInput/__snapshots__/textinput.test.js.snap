// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textinput matches snapshot 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  data-testid="textinputTextId"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-nnbavb"
      >
        <span
          className="notranslate"
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;
